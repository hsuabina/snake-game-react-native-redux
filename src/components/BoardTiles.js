import React, { Component } from 'react'
import Svg, { Rect } from 'react-native-svg'


const EmptyTile = (props) => (
  <Svg width="16" height="16">
    <Rect x="0" y="0" width="16" height="16" fill="#A1C653" />
  </Svg>
)

const SnakeTile = (props) => (
  <Svg width="16" height="16">
    <Rect x="0" y="0" width="16" height="16" fill="black" />
  </Svg>
)

const FoodTile = (props) => (
  <Svg width="16" height="16">
    <Rect x="0" y="0" width="16" height="16" fill="red" />
  </Svg>
)

export { EmptyTile, SnakeTile, FoodTile }
