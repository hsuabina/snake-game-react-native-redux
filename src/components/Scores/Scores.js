import React, { Component, PropTypes } from 'react'
import { View, Text } from 'react-native'

const Scores = ({score}) => (
  <View>
    <Text>SCORE: { score }</Text>
  </View>
)

Scores.propTypes = {
  score: PropTypes.number.isRequired
}


export default Scores
