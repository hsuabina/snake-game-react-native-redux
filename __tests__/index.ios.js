import 'react-native'
import React from 'react'

import { EmptyTile } from 'src/components/BoardTiles/BoardTiles.js'
import Scores from 'src/components/Scores/Scores.js'


describe('<EmptyTile /> from BoardTiles', () => {
  it('matches snapshot', () => {
    const tree = shallow(<EmptyTile />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<SnakeTile /> from SnakeTile', () => {
  it('matches snapshot', () => {
    const tree = shallow(<EmptyTile />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<Scores />', () => {
  it('matches snapshot', () => {
    const tree = shallow(<Scores score={1}/>)
    expect(tree).toMatchSnapshot()
  })
})
