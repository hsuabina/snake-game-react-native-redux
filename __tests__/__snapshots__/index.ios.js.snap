// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyTile /> from BoardTiles matches snapshot 1`] = `
<Svg
  height="16"
  preserveAspectRatio="xMidYMid meet"
  width="16"
>
  <Rect
    fill="#A1C653"
    height="16"
    rx={0}
    ry={0}
    width="16"
    x="0"
    y="0"
  />
</Svg>
`;

exports[`<Scores /> matches snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    SCORE: 
    1
  </Text>
</View>
`;

exports[`<SnakeTile /> from SnakeTile matches snapshot 1`] = `
<Svg
  height="16"
  preserveAspectRatio="xMidYMid meet"
  width="16"
>
  <Rect
    fill="#A1C653"
    height="16"
    rx={0}
    ry={0}
    width="16"
    x="0"
    y="0"
  />
</Svg>
`;
